### PR TITLE
[Google Drive] Record type inclusion for `File` record

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
### Purpose & Motivation
Refactor the bulky record type `File` with inclusion of 2 sub records `FileMetadata` & `GeneratedFileMetadata`

resolves https://github.com/wso2-enterprise/choreo/issues/4893

### CODE REVIEW

File: drive/types.bal [hunk 1]
```diff
@@ -460,15 +485,23 @@ public type CreateFileOptional record {
 # + includePermissionsForView - Specifies which additional view's permissions to include in the response. 
 #                               Only 'published' is supported.
 # + addParents - A comma-separated list of parent IDs to add.
-# + supportsAllDrives -   
+# + supportsAllDrives -  Whether the requesting application supports both My Drives and shared drives. (Default: false)  
+@display {label: "File Metadata Optional Parameters"}
```
Comment (abeykoon) [inline drive/types.bal:489]: There is no need to say `Optional Parameters`. It is acquired by analysing syntax tree. File Metadata is enough?

File: drive/client.bal [hunk 2]
```diff
@@ -305,16 +305,16 @@ public client class Client {
     # 
     # + fileId - ID of the file to be updated
     # + optional - 'UpdateFileMetadataOptional' used to add query parameters to the request
-    # + fileResource - 'File' can added as a payload to change metadata
+    # + fileMetadata - 'FileMetadata' can added as a payload to change metadata
     # + return - If successful, returnsoptionalsFileMetadata `File`. Else returns `error`
     @display {label: "Update File Metadata By Id"}
     remote isolated function updateFileMetadataById(@display {label: "File Id"} string fileId, 
                                                     @display {label: "File Resource"}
```
Comment (abeykoon) [inline drive/client.bal:312]: New File Matadata?

Comment (abeykoon) [reply drive/client.bal:312]: I did not see create file operation in the diff. Please check that also. File metadata should be an optional input.

File: drive/types.bal (hunk 1 above)
Comment (SkNuwanTissera) [reply drive/types.bal:489]: Yeah. We will update like that. Here is a record for `optional query paramters` in file update

File: drive/client.bal (hunk 2 above)
Comment (SkNuwanTissera) [reply drive/client.bal:312]: it was removed in the last commit 
https://github.com/ballerina-platform/module-ballerinax-googleapis.drive/commit/07ae38137a5b20a9a77a1da33d539699966bd575

File: drive/types.bal (hunk 1 above)
Comment (SkNuwanTissera) [reply drive/types.bal]: Updated with the new label

**Review by abeykoon — APPROVED** [review]

**Pull request merged**